### ISSUE TITLE
augmentChunkHash with extract: true breaks hashing

### ISSUE THREAD

**Issue by nstepien** [egoist/rollup-plugin-postcss#315]
My configuration looks like this:
```js
  postcss({
    extract: true,
    minimize: process.env.NODE_ENV === 'production'
  })
```

---

https://rollupjs.org/guide/en/#augmentchunkhash
https://github.com/egoist/rollup-plugin-postcss/blob/master/src/index.js#L140
The `augmentChunkHash` hook is used to augment the hash of chunks (entry js file + its dependencies), not assets, so I don't understand why this is used. It's not even filtering which chunk gets its hash augmented.
This is causing all my js files to have new a hash on every build, even when nothing has changed.

The extracted stylesheet's hash should also not be based on the entry file's hash, it should have its own independent hash.

Removing the `augmentChunkHash` hook completely fixes first problem. `rollup-plugin-postcss` will still need to hash the stylesheet based on its content.

**Comment by nstepien** [comment]
Using the `name` property with `emitFile` is enough to automatically hash that file.

https://rollupjs.org/guide/en/#thisemitfileemittedfile-emittedchunk--emittedasset--string
>Otherwise if a name is supplied, this will be used as substitution for [name] in the corresponding output.chunkFileNames or output.assetFileNames pattern, possibly adding a unique number to the end of the file name to avoid conflicts.

**Comment by rproserpio** [comment]
Yeah, i'm observing the same behavior in my project. This issue breaks caching.

**Comment by rproserpio** [comment]
PR #226 already uses the `emitFile` API with the `name` property, although i believe should be better to pass the actual code without `sourceMappingUrl` as the `source` property to let rollup handle the hash generation, and then append the appropriate`sourceMappingUrl` directly to the asset source in the `bundle`, when we have the sourcemap filename.
If the hash is computed based on the effective output file content, is there any other reason to use `augmentChunkHash`?

**Comment by AlexGalays** [comment]
Did anyone find a fork that fixes these kind of big/completely breaking bugs?

**Comment by rproserpio** [comment]
You can go relatively far using something like [patch-package](https://github.com/ds300/patch-package) to amend specific things until the maintainers have the time to look at the submitted PRs.
To emit an asset together with its sourcemap you can do somenthing like the following (stolen from now deleted code in `rollup-plugin-svelte`):
```js
function emitFileWithSourcemap(bundle, name, code, map) {
	const ref = this.emitFile({
		type: 'asset',
		name: name,
		source: code
	});
	const filename = this.getFileName(ref);
	const mapFilename = `${filename}.map`;
        // TODO handle inline, hidden or disabled sourcemap options
	bundle[filename].source += `\n/*# sourceMappingURL=${path.basename(mapFilename)} */`;
	this.emitFile({
		fileName: mapFilename,
		type: 'asset',
		source: map
	});
}
```
This function lets Rollup handle the naming and the hashing, using its `output.assetFileNames` conf. You can then completely remove `augmentChunkHash`, since the hashing is consistent with the extracted file content.
If you use js dynamic imports that import css you may want to also look at #317 to avoid non deterministic source concatenation.
